clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -5414,6 +5414,8 @@ def note_constraint_normalization_here : Note<
 def note_parameter_mapping_substitution_here : Note<
   "while substituting into concept arguments here; substitution failures not "
   "allowed in concept arguments">;
+def note_building_deduction_guide_here : Note<
+  "while building deduction guide here">;
```

Comment:
Maybe we could make this a bit more descriptive:
```suggestion
  "while building implicit deduction guide first needed here">;
```
(Can we list the template name in this diagnostic too, "... deduction guide for %0 ..."? Might be nice, but I think it'll be pretty obvious most of the time, so it seems non-essential.)